packages/api-queries/src/me-tax-details.ts

```diff
@@ -0,0 +1,21 @@
+import { fetchUserTaxDetails, updateUserTaxDetails } from '@automattic/api-core';
+import { queryOptions, mutationOptions } from '@tanstack/react-query';
+import { queryClient } from './query-client';
+import type { UserTaxFormData } from '@automattic/api-core';
+
+export const userTaxDetailsQuery = () =>
+	queryOptions( {
+		queryKey: [ 'me', 'billing-purchases', 'tax-details' ],
+		queryFn: () => fetchUserTaxDetails(),
+	} );
+
+export const userTaxDetailsMutation = () =>
+	mutationOptions( {
+		mutationFn: updateUserTaxDetails,
+		onSuccess: ( newData ) => {
+			queryClient.setQueryData(
+				userTaxDetailsQuery().queryKey,
+				( oldData ) => oldData && ( { ...oldData, ...newData } as UserTaxFormData )
```

Comment:
This cast isn't necessary, and it actually obscures some of the subtly here I think.

```suggestion
				( oldData ) => oldData && { ...oldData, ...newData }
```

If `oldData === undefined`, it means that we're in an interesting situation where the mutator was called before fetcher was called. In that case, `setQueryData` is a no-op. We don't have enough data to save a complete `UserTaxFormData` to the query cache, and so we don't save anything.

But if `oldData !== undefined`, then we know that `{ ...oldData, ...newData }` will also make a complete `UserTaxFormData` type, and we can safely put it in the query cache without casting.